[Refactor Install]: Helm -> Kustomize for IIS (#1131)
llm-d/llm-d: guides/recipes/scheduler/README.md

# Scheduler Recipes

> llm-d uses the [Gateway API Inference Extension](https://github.com/kubernetes-sigs/gateway-api-inference-extension) scheduler to make intelligent routing decisions for inference requests. There are two deployment modes:

Reviewer: ```suggestion
llm-d uses the [Inference Scheduler](https://github.com/llm-d/llm-d-inference-scheduler) to make intelligent request scheduling decisions for inference requests. There are two deployment modes:
```

## Values Layering

Both modes share a common `base.values.yaml` containing the scheduler image, ports, and common pod selector labels. Feature values (monitoring, tracing) and guide-specific values are layered on top:

```
base.values.yaml                              # shared defaults (this directory)
  + features/monitoring.values.yaml           # optional feature toggles
  + <guide>/scheduler/<guide>.values.yaml     # guide-specific overrides
```

## Standalone (Default)

> Use this when you **do not** want to deploy an external proxy. The standalone chart deploys the scheduler with an embedded Envoy sidecar that handles traffic directly.

Reviewer: ```suggestion
Use this when you **do not** want to deploy a proxy via Kubernetes Gateway APIs. The standalone chart deploys the scheduler EPP with an Envoy sidecar to proxy the traffic directly.
```

**Chart:** `oci://registry.k8s.io/gateway-api-inference-extension/charts/standalone`

```bash
helm install <release-name> \
  oci://registry.k8s.io/gateway-api-inference-extension/charts/standalone \
  -f guides/recipes/scheduler/base.values.yaml \
  -f guides/recipes/scheduler/features/monitoring.values.yaml \
  -f guides/<your-guide>/scheduler/<your-guide>.values.yaml \
  --set provider.name=<gke|istio|none> \
  -n ${NAMESPACE} \
  --version v1.4.0
```

> ## With Proxy

Reviewer: ```suggestion
## With Kubernetes Gateway API
```

> Use this when you want to route traffic through an external proxy (Istio, Agentgateway, etc.) via the Kubernetes Gateway API. This requires:

Reviewer: ```suggestion
Use this when you want to route traffic through a proxy managed by the Kubernetes Gateway API  (e.g., GKE Gateway, Istio, Agentgateway) . This requires:
```

> 1. A proxy control plane installed (see [prereq/gateway-provider](../../prereq/gateway-provider/README.md))

Reviewer: ```suggestion
1. A Gateway control plane installed (see [prereq/gateway-provider](../../prereq/gateway-provider/README.md))
```

> 1. A proxy control plane installed (see [prereq/gateway-provider](../../prereq/gateway-provider/README.md))

Reviewer: We need to keep either `../../prereq/gateway-provider/` or https://github.com/llm-d/llm-d/tree/main/docs/wip-docs-new/guides/gateways, right?

Author: Yeah, one of those, I think wonce the Wip-docs-new "graduates" it will replace that but they will be the same path

2. A Gateway resource created (see [recipes/gateway](../gateway/))

> 3. The inferencepool chart deployed (below)

Reviewer: ```suggestion
2. Creating a Gateway resource (see [recipes/gateway](../gateway/))
3. Deploying the inferencepool chart (below)
```

> The inferencepool chart deploys the scheduler as an Endpoint Picker Pod (EPP). The proxy communicates with the EPP via gRPC ext-proc to determine request routing.

Reviewer: ```suggestion
The inferencepool chart deploys the scheduler as an Endpoint Picker Pod (EPP). The proxy communicates with the EPP via gRPC ext-proc to determine request scheduling.
```
